File: api/risk.js
import { seededRandom, round } from '../lib/sim.js';

export const access = 'public';
export const methods = ['GET'];

export default async function (req, res) {
  const q = req.query || {};
  const route = q.route || q.destination || 'default';
  const rnd = seededRandom(`risk|${route}`);

  const scores = {
    market: round(35 + rnd() * 35, 0),
    port: round(35 + rnd() * 40, 0),
    weather: round(20 + rnd() * 40, 0),
    geopolitical: round(20 + rnd() * 40, 0),
    supply: round(25 + rnd() * 40, 0),
    contract: round(15 + rnd() * 35, 0),
  };
  const overall = round(Object.values(scores).reduce((a, b) => a + b, 0) / Object.keys(scores).length, 0);

  res.json({
    mode: 'simulated',
    route,
    overall_risk: overall,
    scores,
    events: [
      {
        type: 'Australian Cyclone', region: 'Queensland', severity: 'HIGH',
        start: '2026-09-01', end: '2026-09-04', affected_routes: ['Gladstone', 'Hay Point'],
        expected_impact_days: 1.8, source: 'demonstration feed', last_updated: new Date().toISOString(),
      },
    ],
  });
}

File: backend/app/schemas/risk.py
from pydantic import BaseModel


class RiskRequest(BaseModel):
    route_id: str
    origin_country: str
    destination_port: str
    date: str


class RiskScores(BaseModel):
    market: float
    port: float
    weather: float
    geopolitical: float
    supply: float
    contract: float


class RiskResponse(BaseModel):
    mode: str = "rule_based"
    route_id: str
    origin_country: str
    destination_port: str
    destination_port_name: str
    date: str
    market: float
    port: float
    weather: float
    geopolitical: float
    supply: float
    contract: float
    overall: float
    overall_risk: float
    scores: RiskScores
